### DISCUSSION TITLE
Unindexed foreign keys many-to-many table

### DISCUSSION THREAD

**Discussion by thedevdavid** [supabase/supabase#26585] [General]
Performance advisor reports unindexed foreign keys for a junction table. Is this necessary if the junction table has a composite primary key as a unique index?

<img width="1378" alt="image" src="https://github.com/supabase/supabase/assets/15244882/37ec933d-8809-454c-bddc-e821554834bc">

<img width="704" alt="image" src="https://github.com/supabase/supabase/assets/15244882/bd10c61d-4faa-4888-af10-a029a6e665e7">

<img width="812" alt="image" src="https://github.com/supabase/supabase/assets/15244882/ff411649-ef5d-4c90-8380-30dad7de06dc">

**Comment by encima** [comment]
This depends on query usage you have, the index you get "for free" is a composite on `tracks_id` and `playlist_id` but you may have queries that are just querying the `tracks_id` in the table or, possibly, querying both but in a different order such that it is not used.

The advice is not wrong but I do see that this may also result in over-indexing. Not sure this is a bug but we could move it to discussions to see what others think?

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions as @encima mentioned.
